src/reformat.c

```diff
@@ -44,6 +44,10 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
     state->mode = AVIF_REFORMAT_MODE_YUV_COEFFICIENTS;
 
     if (image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) {
+        if (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV444) {
+            return AVIF_FALSE;
+        }
```

Comment:
This restriction should be documented in the commit message. (I agree with this change, but in my experience this restriction is somewhat controversial.) I suggest we move this change to its own pull request so that it can be reviewed separately.